common/token-module/src/lib.rs

```diff
@@ -76,6 +77,7 @@ pub trait TokenModule: fee_estimator_module::FeeEstimatorModule {
                 .set(&default_price_per_gas_unit);
         }
 
+        self.whitelisted_token_mint_burn(&token_id).set(mint_burn_allowed);
```

Comment:
If `mint_burn_allowed` can be set to false, and `whitelisted_token_mint_burn` can store `true` and `false` values, I would suggest renaming it.
`whitelist` suggests that it is allowed.